docs/proposals/distributed-tracing.md

```diff
@@ -0,0 +1,310 @@
+- Compatible with external auto-instrumentation agents
+
+**Future Enhancements:**
+Component owners can independently add detailed instrumentation including:
+- Multiple internal spans for complex operations
+- Component-specific performance metrics and attributes
+- Granular error tracking and debugging information
+- Advanced optimization validation metrics
+
+#### **Component: `llm-d-inference-scheduler (Endpoint Picker Protocol)`**
+
+  * **Component Architecture**: Provides the Endpoint Picker Protocol (EPP) that integrates with the 
+Inference Gateway. It operates as a gRPC service that receives routing requests from the gateway and makes 
+intelligent endpoint selection decisions.
+
+  * **Instrumentation Focus**: This component is responsible for making smart load-balancing and routing decisions,
+applying filtering and scoring algorithms based on awareness of P/D, KV-cache, SLA, and load.
+
+  * **Auto-instrumentation Implementation**:
+    - **Tracing Infrastructure**: Auto-instrumentation using `otel.Tracer("llm-d-inference-scheduler")` without explicit initialization
+    - **Initial Spans**: 
+      - `epp.pd_prerequest`: P/D disaggregation pre-request plugin operation
+    - **Basic Attributes**: 
+      - `llm_d.pd.disaggregation_enabled`: Whether P/D disaggregation is active
+      - `llm_d.pd.prefill_pod_address`: Selected prefill pod address (when applicable)
+      - `operation.outcome`: success/error
+    - **Context Propagation**: Maintains trace context across EPP operations and downstream calls using global tracer
+    - **Benefit**: Establishes EPP visibility in end-to-end traces, P/D disaggregation tracking with zero configuration
+
+  * **Future Enhancement Opportunities**:
+    Component owners can add detailed spans for EPP gRPC requests, pod selection decisions, routing logic, and filter operations. Advanced attributes could include routing decision details, algorithm execution timing, and optimization effectiveness metrics.
+
+#### **Component: `llm-d-kv-cache-manager`**
+
+  * **Instrumentation Focus**: This component manages a global view of KV cache states and localities, for optimizing LLM inference by reusing
+computed key/value attention vectors. It interacts with storage to index KV block availability.
+
+  * **Auto-instrumentation Implementation**:
+    - **Tracing Infrastructure**: Auto-instrumentation using `otel.Tracer("llm-d-kv-cache-manager")` without explicit initialization
+    - **Initial Spans**: `GetPodScores` operation (entry → response)
+    - **Basic Attributes**:
+      - `gen_ai.request.model`: Model identifier
+      - `llm_d.kv_cache.hit_ratio`: Cache hit ratio for the request
+      - `llm_d.kv_cache.pod_count`: Number of pods considered
+      - `operation.outcome`: success/error/timeout
+    - **Context Propagation**: Maintains trace context across cache operations using global tracer
+    - **Redis Instrumentation**: Automatic tracing of Redis operations via `redisotel`
```

Comment:
current [opentelemetry-go-instrumentation](https://github.com/open-telemetry/opentelemetry-go-instrumentation)  can't automatic instrument redis. Only packages listed in https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/main/internal/pkg/inject/offset_results.json can do it.

we can find the instruemt sdk at https://opentelemetry.io/ecosystem/registry/?language=go&s=redis

Reply:
i've removed redis instrumentation from proposal, ty